### PR TITLE
Use rpath with nvidia

### LINKED ISSUE #13: Library path not passed to linker
### Describe the bug
The cuda code does not use the `-Wl,-rpath` flags as nvcc does not recognise this flag. These flags should be passed to the linker using `-Xcompiler`. Without this flag the executable will fail to find any shared libraries which are not on the path

### PR DESCRIPTION
**Pull request by EmilyBourne**
Fixes #13

### CODE REVIEW

**Comment by EmilyBourne** [comment]
@bauom Codacy tests aren't running as they aren't run on the base branch, but I don't think I've changed enough code to break anything. If you are happy with this PR I will remove the additional test trigger and the extra print and merge to your branch

**Comment by EmilyBourne** [comment]
The coverage test is failing because the test is triggered on a push not a pull request. This will be fixed by https://github.com/pyccel/pyccel/pull/1352

File: pyccel/codegen/compiling/compilers.py [hunk 1]
```diff
@@ -333,7 +333,12 @@ def compile_program(self, compile_obj, output_folder, verbose = False):
         # Get compile options
         exec_cmd, includes, libs_flags, libdirs_flags, m_code = \
                 self._get_compile_components(compile_obj, accelerators)
-        linker_libdirs_flags = ['-Wl,-rpath' if l == '-L' else l for l in libdirs_flags]
+        if self._info['exec'] in ('nvcc', 'nvc', 'nvfortran'):
+            linker_libdirs_flags = [f'"-Wl,-rpath,{l[2:]}"' for l in libdirs_flags]
+            linker_libdirs_flags = self._insert_prefix_to_list(linker_libdirs_flags, '-Xcompiler')
```
Comment (bauom) [inline pyccel/codegen/compiling/compilers.py:338]: I tried to use `-Xcompiler` flag but it didn't work what does the `l[2:]` do?

Comment (EmilyBourne) [reply pyccel/codegen/compiling/compilers.py:338]: It gets rid of the "-L". As there are multiple comments being passed Xcompiler is really picky about the format. It has to be
```
-Xcompiler "-Wpath,-rpath,/my/path"
```
The lack of spaces made me change how the `-L` was added. But this does not seem to be necessary now that I see the correct format. I will change for something closer to the original code

Comment (EmilyBourne) [reply pyccel/codegen/compiling/compilers.py]: Fixed [3bad499](https://github.com/pyccel/pyccel-cuda/pull/14/commits/3bad4994545684aaed544abec41aa141107406bf)

**Review by bauom — APPROVED** [review]
the run time test are passing.
everything seems good to me.

**Pull request merged**